Update Microsoft Azure OAuth guide (#1135)
clerk/clerk-docs: docs/authentication/social-connections/microsoft.mdx

---
description: Learn how to allow users to sign into your Clerk app with their Microsoft account using OAuth.
---

# OAuth with Microsoft Azure Entra ID

<TutorialHero
  beforeYouStart={[
    {
      title: "A Clerk application is required.",
      link: "/docs/quickstarts/setup-clerk",
      icon: "clerk",
    },
    {
      title: "A Microsoft Azure account is required.",
      link: "https://azure.microsoft.com/en-us/free/",
      icon: "user-circle",
    }
  ]}
>

- Use Microsoft Azure Entra ID to authenticate users with OAuth.
- Protect your application form [the nOAuth exploit](https://www.descope.com/blog/post/noauth).

</TutorialHero>

Enabling OAuth with Microsoft Azure Entra ID (formerly [Active Directory](https://learn.microsoft.com/en-us/entra/fundamentals/new-name)) allows your users to sign in and sign up to your Clerk application with their Microsoft account.

> Adding social connection with Microsoft Azure to your app with Clerk only requires populating the **Client ID**, **Client Secret** and **Authorized redirect URI** in your instance settings.

Author: ```suggestion
Configuring this feature requires populating the **Client ID**, **Client Secret** and **Authorized redirect URI** in your Clerk app's settings.

```

---

> description: Learn how to set up social connection with Microsoft.

Author: ```suggestion
description: Learn how to allow users to sign into your Clerk app with their Microsoft account using OAuth.
```

---

# OAuth with Microsoft Azure Entra ID

<TutorialHero
  beforeYouStart={[
    {
      title: "A Clerk application is required.",
      link: "/docs/quickstarts/setup-clerk",
      icon: "clerk",
    },
    {
      title: "A Microsoft Azure account is required.",
      link: "https://azure.microsoft.com/en-us/free/",
      icon: "user-circle",
    }
  ]}
>

- Use Microsoft Azure Entra ID to authenticate users with OAuth.
- Protect your application form [the nOAuth exploit](https://www.descope.com/blog/post/noauth).

</TutorialHero>

Enabling OAuth with Microsoft Azure Entra ID (formerly [Active Directory](https://learn.microsoft.com/en-us/entra/fundamentals/new-name)) allows your users to sign in and sign up to your Clerk application with their Microsoft account.

> Adding social connection with Microsoft Azure to your app with Clerk only requires populating the **Client ID**, **Client Secret** and **Authorized redirect URI** in your instance settings.
> 
> To make the development flow as smooth as possible, Clerk uses preconfigured shared OAuth credentials and redirect URIs for development instances - no other configuration is needed.
> 
> For production instances, you will need to generate your own Client ID and Client Secret using your Microsoft Azure account.

Author: ```suggestion
Adding social connection with Microsoft Azure to your app with Clerk only requires populating the **Client ID**, **Client Secret** and **Authorized redirect URI** in your instance settings.

Clerk uses preconfigured shared OAuth credentials and redirect URIs for development instances - no other configuration is needed. For production instances, you must generate your own **Client ID** and **Client Secret** as shown in the instructions in this guide.

```

    1. Under **Redirect URI (Optional)**, select **Web**.
    1. Finally, select **Register** to submit the form.

### Get your client ID and client secret

Once your Microsoft Entra ID app is created, or once you select your app from the Entra ID dashboard, you'll be taken to its **Overview**.

1. From your app's overview, copy the **Application (client) ID** and save it somewhere secure. It's required connecting your Microsoft Entra ID app to your Clerk application.
1. On this same page, under **Client credentials**, select **Add a certificate or secret** to generate a Client Secret.
1. You'll be taken to the **Certificate & secrets** tab, where you can select the **New client secret** button. In the modal that appears, add a description and set an expiration time for your secret.
    > [!IMPORTANT]
    > When your secret expires, your social connection will stop working until you generate a new client secret and add it to your Clerk app.
1. Copy the test in your new client secret's **Value** field and save it somewhere secure. You'll add it to your Clerk application later, alongside your client ID.

> ### Connect Entra ID app and get your redirect URI

Reviewer: ```suggestion
### Connect your Entra ID app and get your redirect URI

```

1. In another tab, visit the Clerk Dashboard.
1. In the sidebar, go to **User & Authentication > [Social Connections](https://dashboard.clerk.com/last-active?path=user-authentication/social-connections)**. In the list of **Auth Providers**, enable **Microsoft**.
1. Ensure that both **Enable for sign-up and sign-in** and **Use custom credentials** are toggled on. Then:
    - Under **Client ID**, add the value you copied from **Application (client) ID** in the Entra ID dashboard.
    - Under **Client Secret**, add the client secret value you generated in the Entra ID dashboard.
    - Copy the **Authorized redirect URI**. You need it for the final step to configure your Entra ID app.
    - Select **Save**.

### Enable OpenID

> For OAuth to function, Clerk requires an `id_token` from Microsoft Entra ID.

Reviewer: We mention `id_token` here but the user doesn't use this, do they? I'm not sure why we mention it.

We can just say something like "To connect your Clerk app to your Microsoft app, you must set the **Authorized redirect URI** in your Microsoft dashboard.

1. Return to the tab where your Microsoft Entra ID application dashboard is open.
1. In the sidebar, select **Authentication**, then select **Add a platform**.
1. In the **Redirect URIs** field of the modal that appears, add your Clerk application's authorized redirect URI. Do the same in the **Front-channel logout URL** field.

### Test your OAuth

> The simplest way to test your OAuth is to visit your Clerk application's account portal, which Clerk creates for all applications.

Reviewer: Two suggestions here based on where we want to say "out-of-the-box"

```suggestion
The simplest way to test your OAuth is to visit your Clerk application's [Account Portal](/docs/account-portal/overview), which is available for all Clerk applications out-of-the-box.

```

```suggestion
The simplest way to test your OAuth is to visit your Clerk application's [Account Portal](/docs/account-portal/overview), which is available out-of-the-box for all Clerk applications.

```
